portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -33,12 +33,26 @@
 public class DataServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String commentcount = request.getParameter("count");
+    int maxcomments;
+    try {
+        maxcomments = Integer.parseInt(commentcount);
+    } catch(NumberFormatException e){
+        // Display nothing for now.
+        maxcomments = 0;
+    }
     Query query = new Query("Show").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
consider extracting this datastore stuff into a helper:  

List<String> fetchShows(Datastore datastore);

(In general it's a good idea to have all your datastore stuff in a single library, which you'll probably want to do for the capstone project.  Typical library name might be KindStore, or in this case the ShowsStore.  Beyond just organizing your logic into one place, there are practical benefits since it could let you define a https://en.wikipedia.org/wiki/Data_access_object if you wanted to later on.)

Reply:
Will note for capstone.  For now I'm leaving it in the function since I wont be using it anywhere else.